01-data-structures/04-hashes-part-1/hashes-1-answers.txt

```diff
@@ -0,0 +1 @@
+1. Doubling the size of the array in the class is problematic for a number of reasons. First it doesn’t necessarily prevent another collision, to which the only solution is to again double it (which also doesn’t guarantee a collision-free environment,) so it may take quite a few doubles before we actually end up with adequate locations for all our hash items. This leads to a second problem in terms of space: It would seem to me that doubling the array creates a lot of unnecessary unused space, additionally the actual creation of the array as well as the demands of shifting and reorganizing the arrays various index at each doubling would also have particularly excessive memory/processing demands. 
```

Comment:
This is a good way of looking at it. Simply resizing in general does not solve many of the underlying issues. Doubling in particular is bad due to the factorability of powers of the same base (in this case, two). Doubling the size of the array repeatedly will not change the collision points at all, but rather may delay the collision from occurring a little bit. That is why prime numbers are better to use.